src/lib.rs

```diff
@@ -715,6 +715,24 @@ impl<const LEN: usize, TItem: FromMmioPtr<TMmio = TMmio>, TMmio: Mmio + Copy> Ar
             ptr: self.ptr,
         }
     }
+
+    #[inline(always)]
+    pub fn get_sub_array<const NEW_LEN: usize>(
+        &self,
+        start_offset: usize,
```

Comment:
nit: should this be an array index instead of a memory offset? Seems like taking a memory offset makes it possible/easier to get unaligned arrays. I assume that's undesirable.

Reply:
It is an index. Renamed to start_index to make it clearer.

FYI: from [pointer::add() docs](https://doc.rust-lang.org/std/primitive.pointer.html#method.add)

> count is in units of T; e.g., a count of 3 represents a pointer offset of 3 * size_of::<T>() bytes.

Reply:
Oh cool, thanks!